src/Domain/Value/Resolver/RelationCollection.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of Storyblok-Api.
+ *
+ * (c) SensioLabs Deutschland <info@sensiolabs.de>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Storyblok\Api\Domain\Value\Resolver;
+
+/**
+ * @author Silas Joisten <silasjoisten@proton.me>
+ *
+ * @implements \IteratorAggregate<int, Relation>
+ */
+final class RelationCollection implements \Countable, \IteratorAggregate
+{
+    /**
+     * @var list<Relation>
+     */
+    private array $items = [];
+
+    /**
+     * @param list<Relation> $items
+     */
+    public function __construct(
+        array $items = [],
+    ) {
+        foreach ($items as $item) {
+            $this->add($item);
+        }
+    }
+
+    /**
+     * @return \Traversable<int, Relation>
+     */
+    public function getIterator(): \Traversable
+    {
+        return new \ArrayIterator($this->items);
+    }
+
+    public function count(): int
+    {
+        return \count($this->items);
+    }
+
+    public function add(Relation $relation): void
+    {
+        if ($this->has($relation)) {
+            return;
+        }
+
+        $this->items[] = $relation;
+    }
+
+    public function has(Relation $relation): bool
+    {
+        foreach ($this->items as $item) {
+            if ($item->value === $relation->value) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    public function remove(Relation $relation): void
```

Comment:
Not sure we need a remove

Reply:
It is handy in some cases. i used it on our website in BlogPostList (for filters in this case)